assembl/graphql/schema.py

```diff
@@ -1240,143 +1260,22 @@ def mutate(root, args, context, info):
                 subject=subject_langstring,
                 body=body_langstring,
                 creator_id=user_id,
-                body_mime_type=u'text/html'
+                parent=in_reply_to_post
             )
             new_post.guess_languages()
-            # TODO
-            # new_post.body_mime_type = 'text/html'
             db = new_post.db
             db.add(new_post)
+            idea_post_link = models.IdeaRelatedPostLink(
+                creator_id=user_id,
+                content=new_post,
+                idea=in_reply_to_idea
+            )
+            db.add(idea_post_link)
             db.flush()
-            if in_reply_to_post:
-                new_post.set_parent(in_reply_to_post)
-            elif in_reply_to_idea:
-                # don't create IdeaRelatedPostLink when we have both
-                # in_reply_to_post and in_reply_to_idea
-                idea_post_link = models.IdeaRelatedPostLink(
-                    creator_id=user_id,
-                    content=new_post,
-                    idea=in_reply_to_idea
-                )
-                db.add(idea_post_link)
-
-            db.flush()
-            new_post.db.expire(new_post, ['idea_content_links_above_post'])
```

Comment:
Because it seems to be already calculated at the creation of the post, then I add the idea_post_link, and I need this property to recalculate when returning the post. And the flush doesn't do it because expire_on_flush=False.